File: benchmarkDS/Vanilla/OPENAI_gpt-5-2025-08-07/Lse125.py
import sys

def trim_trailing(s):
    return None if s is None else s.rstrip()

if __name__ == "__main__":
    if len(sys.argv) > 1:
        print(trim_trailing(sys.argv[1]))
    else:
        tests = [
            "hello   ",
            "test\t\t",
            "line\n",
            " mix  \t\n",
            "no-trim"
        ]
        for t in tests:
            print(trim_trailing(t))

File: benchmarkDS/ourMethod/OPENAI_gpt-5-2025-08-07/Lse125.java
import java.nio.charset.StandardCharsets;

public class Lse125 {

    public static String trimTrailingWhitespace(String input) {
        if (input == null) {
            return "";
        }
        int end = input.length() - 1;
        while (end >= 0 && Character.isWhitespace(input.charAt(end))) {
            end--;
        }
        return input.substring(0, end + 1);
    }

    public static void main(String[] args) {
        // If a command-line argument is provided, trim trailing whitespace and print it.
        if (args.length > 0) {
            String trimmed = trimTrailingWhitespace(new String(args[0].getBytes(StandardCharsets.UTF_8), StandardCharsets.UTF_8));
            System.out.println(trimmed);
        }

        // Five test cases
        String[] tests = new String[] {
            "hello",
            "hello   ",
            "line\n",
            "tabs\t\t",
            "mix \t\n\r"
        };

        for (int i = 0; i < tests.length; i++) {
            String result = trimTrailingWhitespace(tests[i]);
            System.out.println("TEST " + (i + 1) + ": [" + result + "]");
        }
    }
}